src/Query/Builder.php

```diff
@@ -1763,9 +1767,9 @@ public function orWhereIntegerNotInRaw($column, $values, $boolean = 'and')
         throw new BadMethodCallException('This method is not supported by MongoDB');
     }
 
-    private function aliasIdForQuery(array $values): array
+    private function aliasIdForQuery(array $values, bool $root = true): array
     {
-        if (array_key_exists('id', $values)) {
+        if (array_key_exists('id', $values) && ($root || $this->getConnection()->getRenameEmbeddedIdField())) {
```

Comment:
Perhaps I am overly cautious, but I was wondering whether repeatedly calling `$this->getConnection()` could posssibly introduce a noticeable performance penalty worth worrying about? (This method appears to get called from inside loops, after all.) Or is this just a simple property lookup?

Reply:
You're right, it's better to use the property since it's available.